remediation_worker/jobs/azure_key_vault_logging_for_keyvault_enabled/azure_key_vault_logging_for_keyvault_enabled.py

```diff
@@ -106,16 +127,349 @@ def parse(self, payload):
+            account_name=name,
+            parameters=BlobServiceProperties(
+                delete_retention_policy=DeleteRetentionPolicy(enabled=True, days=7)
+            ),
+        )
+        return stg_account
+
+    def update_storage_account_encryption(
+        self, storage_client, resource_group_name, stg_name, key_name, vault_uri
+    ):
+        """Updates Storage Account Encryption for a Storage Account.
+        :param storage_client: Instance of the Azure StorageManagementClient.
+        :param resource_group_name: The name of the resource group.
+        :param stg_name: The Storage Account name.
+        :param key_name: Name of the Key to encrypt the Storage Account with.
+        :param vault_uri: Key Vault uri in which the Key exists.
+        :type resource_group_name: str
+        :type stg_name: str
+        :type key_name: str
+        :type vault_uri: str
+        """
+        logging.info("    Encrypting Storage Account with Customer Managed Key")
+        logging.info("    executing storage_client.storage_accounts.update")
+        logging.info(f"      resource_group_name={resource_group_name}")
+        logging.info(f"      account_name={stg_name}")
+        logging.info(f"      key_vault_uri={vault_uri}")
+        logging.info(f"      key_name={key_name}")
+        storage_client.storage_accounts.update(
+            resource_group_name=resource_group_name,
+            account_name=stg_name,
+            parameters=StorageAccountUpdateParameters(
+                encryption=Encryption(
+                    key_source=KeySource.MICROSOFT_KEYVAULT,
+                    key_vault_properties=KeyVaultProperties(
+                        key_name=key_name, key_vault_uri=vault_uri,
+                    ),
+                ),
+            ),
+        )
+
+    def check_stg_account(self, storage_client, region, name, resource_group_name):
+        """Checks For the existence of the Storage Account created by CHSS
+        :param storage_client: Instance of the Azure StorageManagementClient.
+        :param region: The location in which the storage account exists.
+        :param name: The Storage Account name.
+        :param resource_group_name: The name of the resource group.
+        :type region: str
+        :type name: str
+        :type resource_group_name: str
+        :returns: StorageAccount object
+        """
+        storage_accounts_paged: ItemPaged[
+            StorageAccountListResult
+        ] = storage_client.storage_accounts.list()
+        storage_accounts_list: List[dict] = list(storage_accounts_paged)
+        for stg_account in storage_accounts_list:
+            stg_id = stg_account.id
+            stg_components = stg_id.split("/")
+            resource_grp = stg_components[4]
+            if (
+                stg_account.name == name
+                and stg_account.location == region
+                and resource_grp == resource_group_name
+            ):
+                return stg_account
```

Comment:
stroage_account = None at the start

Run the for loop and in the for loop, storage_account = stg_account

return storage_account